assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3819,8 +3834,25 @@ protected String build_instructor_grade_submission_context(VelocityPortlet portl
             state.setAttribute(NEW_ASSIGNMENT_CHECK_ANONYMOUS_GRADING, assignmentService.assignmentUsesAnonymousGrading(a));
 
             boolean allowToGrade = true;
-            if (StringUtils.isNotBlank(a.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT))) {
+            String assignmentAssociateGradebook = a.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT);
+            if (StringUtils.isNotBlank(assignmentAssociateGradebook)) {
                 String gradebookUid = toolManager.getCurrentPlacement().getContext();
+                // If the assignment reference is not equal to the associated gradebook item, then a custom gb item is being used
+                // S2U-34 In the 22x version this check was looking if the retrieved object from the gradingservice is null, now we check if an exception is thrown
+                if (!assignmentRef.equals(assignmentAssociateGradebook)) {
+                    try {
+                        org.sakaiproject.grading.api.Assignment gbAssignment = gradingService.getAssignment(gradebookUid,
+                                assignmentAssociateGradebook);
+
+                        Long associateGradebookAssignmentId = gbAssignment.getId();
+                        context.put("associatedToGbItem", true);
+                        context.put("associatedToGbEntityId", associateGradebookAssignmentId);
+                    } catch (AssessmentNotFoundException e) {
```

Comment:
wtf

Reply:
AssessmentNotFoundException has been in edu-services forever. To the grading service, Assignment = Assessment. Go figure.